src/InnerHit/NestedInnerHit.php

```diff
@@ -145,14 +147,14 @@ public function toArray()
     {
         $out = array_filter(
             [
-                'query' => $this->getQuery()->toArray(),
+                'query' => $this->getQuery() ? $this->getQuery()->toArray() : null,
```

Comment:
Yep, you are right when I wrote a comment I still was thinking that $query has to be mandatory and interface prevents of missing function..

Ok, we can leave here as you did.